### PR TITLE
Add ADR to select an authentication approach

### PR DESCRIPTION
**Pull request by jazairi**
#### Why these changes are being introduced:

We will need authentication in TACOS for the staff UI.

#### Relevant ticket(s):

* [TCO-28](https://mitlibraries.atlassian.net/browse/TCO-28)

#### How this addresses that need:

This adds an ADR to select an authentication approach from among Devise, Clearance, Sorcery, AuthLogic, and a custom solution using built-in Rails security features.

#### Side effects of this change:

The decision we make will affect how we implement authentication.

[TCO-28]: https://mitlibraries.atlassian.net/browse/TCO-28?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: docs/architecture-decisions/0007-use-devise-for-authentication.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+
+## Status
+
+Accepted
+
+## Context
+
+We will need to authenticate users in TACOS to help manage access to the staff interface. The ideal authentication
+approach would involve Omniauth integration, as IS&T recently implemented Okta, and
+[a strategy already exists](https://github.com/omniauth/omniauth-okta) to authenticate in Okta using Omniauth.
```
Comment (JPrevost) [inline docs/architecture-decisions/0007-use-devise-for-authentication.md:13]: I don't think we should use that omniauth integration as it seems unsupported, https://github.com/omniauth/omniauth_openid_connect is more likely the path forward.

File: docs/architecture-decisions/0007-use-devise-for-authentication.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+implement due to their lack of Omniauth integration.
+
+Authlogic feels equally complex as Devise, albeit more transparent. It also seems not to be especially well maintained.
+
+A custom solution would be an interesting way leverage built-in Rails functionality while reducing our reliance on
+third-party libraries. However, it is difficult to justify considering the additional risk and development time.
+
+Ultimately, while all of these alternatives are appealing, there is a distinct advantage to choosing a gem that "just
+works" and is widely adopted. Much like our
+[decision to use CanCanCan for authorization](https://github.com/MITLibraries/tacos/blob/main/docs/architecture-decisions/0005-use-cancancan-for-authorization.md), I might not feel like Devise is the best authentication gem, but I have
```
Comment (JPrevost) [inline docs/architecture-decisions/0007-use-devise-for-authentication.md:96]: I think this will need an updated URL as that linked ADR is changing to 0006

**Review by JPrevost — APPROVED** [review]
Solid write up, thanks!

I support the decision in this ADR as written 👍🏻

**Review by matt-bernhardt — APPROVED** [review]
Thanks for this write-up. While I'm not separately familiar with the options for authentication gems, I agree with the framing you've provided, and support the decision you reach based on that framing.

While in general I like to build things ourselves, there are certain domains where it feels better to trust in the experience of others. Authentication is one of those domains.

:shipit:

**Pull request merged**